snuba/cli/migrations.py

```diff
@@ -39,3 +41,41 @@ def list() -> None:
             click.echo(f"[{symbol}]  {migration_id}{in_progress_text}{blocking_text}")
 
         click.echo()
+
+
+@migrations.command()
+@click.option("--group", required=True, help="Migration group")
+@click.option("--migration-id", required=True, help="Migration ID")
+@click.option("--force", is_flag=True)
+def run(group: str, migration_id: str, force: bool) -> None:
+    """
+    Runs a single migration.
+    The --force option must be passed in order to run blocking migrations.
+
+    Migrations that are already in an in-progress or completed status will not be run.
+    """
+    runner = Runner()
+    migration_group = MigrationGroup(group)
+    migration_key = MigrationKey(migration_group, migration_id)
+
+    pending_migrations = runner.get_pending_migrations(migration_group)
+
+    if migration_id not in [m.migration_id for m in pending_migrations]:
+        try:
+            get_group_loader(migration_group).load_migration(migration_id)
+        except MigrationDoesNotExist as e:
+            raise click.ClickException(str(e))
+
+        raise click.ClickException(
+            f"{migration_group.value}: {migration_id} is already completed"
+        )
+
+    next_migration_id = pending_migrations[0].migration_id
+
+    if next_migration_id != migration_id:
+        raise click.ClickException(f"Earlier migrations need to be completed first")
```

Comment:
The integrity of the database depends on this check (to ensure we run migrations in the right order, which implies the consistency of the final state). I would enforce this in the runner for everyone running migrations not in the script, which is just the interface.
We can have a specific method to force running a specific migration without checks but it should not be the default.